src/wallet.py

```diff
@@ -0,0 +1,86 @@
+import json
+from json.decoder import JSONDecodeError
+
+import structlog
+from nacl.encoding import HexEncoder
+from nacl.signing import SigningKey
+import base64
+from cryptography.fernet import Fernet
+
+
+logger = structlog.get_logger(__name__)
+
+class Wallet:
+    def __init__(self): 
+        self.privatekey = ""
+        self.publickey = ""
+    
+    def generate_wallet():
+        password = input("\nSe necesita una contraseña, escriba una: ")   #Proposing a change 
+        passws = password.encode('utf-8') #It is coded to be supported
+        #Fernet asks for a length of 32 bytes, to the password used add as many '0's to complete it.
+            
+        pwd = base64.b64encode(passws.ljust(32))
+        f = Fernet(pwd)
+        private_key = SigningKey.generate()
+        public_key = private_key.verify_key
+        payload = {
```

Comment:
Up to you, but maybe I would only store in the file the encrypted `private_key`. The `public_key` can be derived from the private key.

Reply:
It could be done, I don't doubt it, it's just a matter of including the attribute when it's going to be used. While I finish polishing the file, it serves me to check that it works well, it has a customisation at the beginning to identify it. When everything works well, it is removed and created when needed with the customisation.